qase-javascript-commons/src/reporters/abstract-reporter.ts

```diff
@@ -68,6 +62,32 @@ export abstract class AbstractReporter implements ReporterInterface {
     return this.results;
   }
 
+  /**
+   * @param {TestResultType} result
+   */
+  public addTestResult(result: TestResultType) {
+    if (result.testops_id === null || !Array.isArray(result.testops_id)) {
+      this.results.push(result);
+      return;
+    }
+
+    // if we have multiple ids, we need to create multiple test results and set duration to 0 for all but the first one
+    let firstCase = true;
+
+    for (const id of result.testops_id) {
+      const testResultCopy = { ...result };
```

Comment:
`{ ...result }` makes a deep copy?

Reply:
We could use slightly different logic here to avoid copying for the first id in the list. And in most cases there will be just one id.

```
if (firstCase) {
  this.results.push(result)
  firstCase = false;
} else {
  const testResultCopy = { ...result};
  testResultCopy.testops_id = id;
  testResultCopy.id = uuidv4();
  testResultCopy.execution.duration = 0;
  this.results.push(testResultCopy)
}
```

Reply:
yep